src/app/(sidebar)/transaction/sign/components/Overview.tsx

```diff
@@ -0,0 +1,286 @@
+
+import { useStore } from "@/store/useStore";
+
+import { transactionSigner } from "@/helpers/transactionSigner";
+
+import { validate } from "@/validate";
+
+import { Box } from "@/components/layout/Box";
+import { MultiPicker } from "@/components/FormElements/MultiPicker";
+import { TextPicker } from "@/components/FormElements/TextPicker";
+import { WithInfoText } from "@/components/WithInfoText";
+import { ValidationResponseCard } from "@/components/ValidationResponseCard";
+import { XdrPicker } from "@/components/FormElements/XdrPicker";
+
+const MIN_LENGTH_FOR_FULL_WIDTH_FIELD = 30;
+
+export const Overview = () => {
+  const { network, transaction } = useStore();
+  const { sign, updateSignActiveView, updateSignedTx, resetSign } = transaction;
+
+  const [secretInputs, setSecretInputs] = useState<string[]>([""]);
+  const [signedTxSuccessMsg, setSignedTxSuccessMsg] = useState<string>("");
+
+  // @TODO bip path
+  const [bipPath, setBipPath] = useState<string>("");
+  const [bipPathErrorMsg, setBipPathErrorMsg] = useState<string>("");
+
+  const HAS_SECRET_KEYS = secretInputs.some((input) => input !== "");
+
+  useEffect(() => {
+    if (!sign.importTx) {
+      updateSignActiveView("import");
+    }
+  }, []);
+
+  const onUpdateSecretInputs = (val: string[]) => {
+    setSecretInputs(val);
+    updateSignedTx("");
+  };
+
+  const addSignature = () => {
+    setSecretInputs([...secretInputs, ""]);
+  };
+
+  const signTxWithSecretKeys = (
+    txXdr: string,
+    signers: string[],
+    networkPassphrase: string,
+  ) => {
+    const { xdr, message } = transactionSigner.secretKeys({
+      txXdr,
+      signers,
+      networkPassphrase,
+    });
+
+    if (xdr) {
+      updateSignedTx(xdr);
+    }
+    setSignedTxSuccessMsg(message);
+  };
+
+  const REQUIRED_FIELDS = [
+    {
+      label: "Signing for",
+      value: network.passphrase,
+    },
+    {
+      label: "Transaction Envelope XDR",
+      value: sign.importXdr,
+    },
+    {
+      label: "Transaction Hash",
+      value: sign.importTx!.hash().toString("hex"),
+    },
+  ];
+
+  let mergedFields;
+
+  if (sign.importTx instanceof FeeBumpTransaction) {
+    mergedFields = [...REQUIRED_FIELDS, ...FEE_BUMP_TX_FIELDS(sign.importTx)];
+  } else if (sign.importTx) {
+    mergedFields = [...REQUIRED_FIELDS, ...TX_FIELDS(sign.importTx)];
+  }
+
+  return (
+    <>
+      <div className="SignTx__Overview">
+        <div className="PageHeader">
+          <Text size="md" as="h1" weight="medium">
+            Transaction Overview
+          </Text>
+
+          <Button
+            size="md"
+            variant="error"
+            icon={<Icon.RefreshCw01 />}
+            iconPosition="right"
+            onClick={() => {
+              resetSign();
+            }}
+          >
+            Clear and import new
+          </Button>
+        </div>
+
+        <Card>
+          <div className="SignTx__FieldViewer">
+            {mergedFields!.map((field) => {
```

Comment:
Just to be safe, let's use `?` instead of "!" in case we don't get `mergedFields`.